src/ethereum_spec_tools/evm_tools/b11r/b11r_types.py

```diff
@@ -0,0 +1,167 @@
+            assert stdin is not None
+            txs_data = stdin["txs"]
+        else:
+            with open(options.input_txs) as f:
+                txs_data = json.load(f)
+
+        self.transactions = rlp.decode(hex_to_bytes(txs_data))
+
+        # Parse ommers
+        if options.input_ommers == "stdin":
+            assert stdin is not None
+            ommers_data = stdin["ommers"]
+        else:
+            with open(options.input_ommers) as f:
+                ommers_data = json.load(f)
+
+        self.ommers = []
+        for ommer in ommers_data:
+            decoded_ommer = rlp.decode(hex_to_bytes(ommer))
+            self.ommers.append(decoded_ommer[0])  # ommer[0] is the header
+
+        # Parse withdrawals
+        if options.input_withdrawals is None:
+            self.withdrawals = []
+            return
+
+        if options.input_withdrawals == "stdin":
+            assert stdin is not None
+            withdrawals_data = stdin["withdrawals"]
+        else:
+            with open(options.input_withdrawals) as f:
+                withdrawals_data = json.load(f)
+
+        self.withdrawals = []
+        for wd in withdrawals_data:
+            self.withdrawals.append(
+                (
+                    parse_hex_or_int(wd["index"], U64),
+                    parse_hex_or_int(wd["validatorIndex"], U64),
+                    Bytes20(hex_to_bytes(wd["address"])),
+                    parse_hex_or_int(wd["amount"], Uint),
+                )
+            )
+
+
+class Header:
+    """
+    A class representing a block header.
+    """
+
+    parent_hash: Hash32
+    ommers_hash: Hash32
+    coinbase: Bytes20
+    state_root: Hash32
+    transactions_root: Hash32
+    receipt_root: Hash32
+    bloom: Bytes256
+    difficulty: Uint
+    number: Uint
+    gas_limit: Uint
+    gas_used: Uint
+    timestamp: U256
+    extra_data: Bytes
+    mix_digest: Bytes32
+    nonce: Bytes8
+    base_fee_per_gas: Optional[Uint]
+    withdrawals_root: Optional[Hash32]
+
+    def __init__(self, options: Any, body: Body, stdin: Any = None):
+        if options.input_header == "stdin":
+            assert stdin is not None
+            data = stdin["header"]
+        else:
+            with open(options.input_header) as f:
+                data = json.load(f)
+
+        self.parent_hash = Hash32(hex_to_bytes(data["parentHash"]))
+        try:
+            self.ommers_hash = Hash32(hex_to_bytes(data["ommersHash"]))
+        except KeyError:
+            self.ommers_hash = keccak256(rlp.encode(body.ommers))
+
+        self.coinbase = Bytes20(
+            hex_to_bytes(
+                data.get("miner", "0x0000000000000000000000000000000000000000")
+            )
+        )
+        self.state_root = Hash32(hex_to_bytes(data["stateRoot"]))
+        self.transactions_root = Hash32(
+            hex_to_bytes(
+                data.get(
+                    "transactionsRoot",
+                    "0x56e81f171bcc55a6ff8345e692c0f86"
```

Comment:
This should be a named constant. Especially since it appears multiple times.

Reply:
Updated